lib/modules/dosomething/dosomething_user/dosomething_user.module

```diff
@@ -524,6 +517,52 @@ function dosomething_user_new_user($form, &$form_state) {
 }
 
 /**
+ * Sets the user global attributes and sends them to Northstar.
+ *
+ * @param object $account
+ *   The account to return value for. If NULL, uses global $user.
+ * @param string $country_code
+ *   The country code where the user currently resides. If NULL, uses
+ *   the headers in current request.
+ * @param string $language
+ *   The new language to set on the user. If NULL, gets the matching
+ *   language or defaults.
+ */
+function dosomething_user_set_global_attributes($user = NULL, $country_code = NULL, $language = NULL) {
+  if ($user == NULL) {
+    global $user;
+  }
+  if ($country_code == NULL) {
+    $country_code = dosomething_settings_get_geo_country_code();
+    // If no Fastly headers set, use US EN
+    if ($country_code == NULL) {
+      $language = 'en';
+    }
+  }
+  // If the language wasn't specified in the function call, determine the user language
+  if ($language == NULL) {
+    $converted_language = dosomething_global_convert_country_to_language($country_code);
+    $language = 'en-global';
+
+    // If the country in the headers matched to a language in our strongarm
+    if ($converted_language != NULL) {
+
+      // Verify the language is installed on the system,
+      $system_languages = language_list();
+      foreach ($system_languages as $lang_key => $system_lang) {
+        if ($converted_language == $lang_key) {
+          $language = $converted_language;
+          break;
+        }
+      }
+    }
+  }
+
+  user_save($user, ['language' => $language]);
```

Comment:
Is the `$country_code` value suppose to be included in the `user_save($user, ['language' => $language]);` ? 

Perhaps looking up the user country should always be done based on the user language setting? This seems problematic long run if we have more than one country with the same language but need know the users actual country.